embassy-stm32/build.rs

```diff
@@ -379,6 +413,21 @@ fn main() {
         }
     }
 
+    let mut refcount_mod = TokenStream::new();
+    for refcount_static in refcount_statics {
+        refcount_mod.extend(quote! {
+            pub(crate) static #refcount_static: AtomicU8 = AtomicU8::new(0);
+        });
+    }
+
+    g.extend(quote! {
+        mod refcount_statics {
+            use atomic_polyfill::AtomicU8;
```

Comment:
`atomic_polyfill` is deprecated, I would try to not introduce more uses of it.

Since the refcounting always happens within a critical section, you can use `static mut` with no atomic instructions at all.